### PR TITLE
Add list, datepicker and textfield component guides

### PR DESCRIPTION
**Pull request by prakhargupta1**
List: https://deploy-preview-2827--mui-toolpad-docs.netlify.app/toolpad/components/list/
DatePicker: https://deploy-preview-2827--mui-toolpad-docs.netlify.app/toolpad/components/date-picker/
TextField: https://deploy-preview-2827--mui-toolpad-docs.netlify.app/toolpad/components/text-field/

- [x] Information about isInvalid prop needs to be added. It's use is unclear as discussed here: https://github.com/mui/mui-toolpad/issues/2826
- [x] Clicking on datepicker and textfiled throws an error
- [x] Remove demo for list component

### CODE REVIEW

**Comment by prakhargupta1** [comment]
Facing some webpack issue. @Janpot Can you take a look? Thanks

Screenshot 2024-02-27 at 11 52 04 AM

**Comment by bharatkashyap** [comment]
> Facing some webpack issue. @Janpot Can you take a look? Thanks
> 
> Screenshot 2024-02-27 at 11 52 04 AM

There might be an extra space somewhere in the `Datepicker.md` file - this should go away if you remove it

**Comment by prakhargupta1** [comment]
> There might be an extra space somewhere in the Datepicker.md file - this should go away if you remove it

Just checked, found none. It was working today when I was on yarn. The other two similar files datagrid.md, button.md are working fine.
The failing lint test also mention about some module issue: `Unable to resolve path to module.` Maybe these are related.

File: docs/data/toolpad/components/datepicker/datepicker.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# DatePicker
+
+<p class="description">Learn about the datepicker component and it's usage in Toolpad.</p>
```
Comment (Janpot) [inline docs/data/toolpad/components/datepicker/datepicker.md:3]:
```suggestion
<p class="description">Learn about the datepicker component and its usage in Toolpad.</p>
```

File: docs/data/toolpad/components/list/list.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# List
+
+<p class="description">Learn about the list component and it's usage in Toolpad.</p>
```
Comment (Janpot) [inline docs/data/toolpad/components/list/list.md:3]:
```suggestion
<p class="description">Learn about the list component and its usage in Toolpad.</p>
```

File: docs/data/toolpad/components/textfield/textfield.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
+# Textfield
+
+<p class="description">Learn about the textfield component and it's usage in Toolpad.</p>
```
Comment (Janpot) [inline docs/data/toolpad/components/textfield/textfield.md:3]:
```suggestion
<p class="description">Learn about the textfield component and its usage in Toolpad.</p>
```

File: docs/pages/toolpad/components/datepicker.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+import MarkdownDocs from 'docs/src/modules/components/MarkdownDocs';
+import * as pageProps from '../../../data/toolpad/components/datepicker/datepicker.md?@mui/markdown';
```
Comment (Janpot) [inline docs/pages/toolpad/components/datepicker.js:3]:
```suggestion
import * as pageProps from '../../../data/toolpad/components/datepicker/datepicker.md?@muiMarkdown';
```

File: docs/pages/toolpad/components/list.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+import MarkdownDocs from 'docs/src/modules/components/MarkdownDocs';
+import * as pageProps from '../../../data/toolpad/components/list/list.md?@mui/markdown';
```
Comment (Janpot) [inline docs/pages/toolpad/components/list.js:3]:
```suggestion
import * as pageProps from '../../../data/toolpad/components/list/list.md?muiMarkdown';
```

File: docs/pages/toolpad/components/textfield.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+import MarkdownDocs from 'docs/src/modules/components/MarkdownDocs';
+import * as pageProps from '../../../data/toolpad/components/textfield/textfield.md?@mui/markdown';
```
Comment (Janpot) [inline docs/pages/toolpad/components/textfield.js:3]:
```suggestion
import * as pageProps from '../../../data/toolpad/components/textfield/textfield.md?muiMarkdown';
```

**Comment by Janpot** [comment]
I pushed a fix

File: docs/data/toolpad/components/datepicker/DatePickerFormat.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { DatePicker } from '@mui/toolpad-components';
+import { Stack } from '@mui/material';
+
+const TOOLPAD_PROPS1 = {
+  size: 'small',
+  variant: 'outlined',
+  label: 'MM/dd/yyyy',
+  format: 'MM/dd/yyyy',
+  // defaultValue: '2017-05-24',
```
Comment (Janpot) [inline docs/data/toolpad/components/datepicker/DatePickerFormat.js:10]:
```suggestion
```

avoid commented out code

File: docs/pages/toolpad/components/datepicker.js
[the remark this answers is not in the document]
Comment (prakhargupta1) [reply docs/pages/toolpad/components/datepicker.js:1]: Yes, the file names are data-grid, date-picker, and file-picker. Ok, so I'll follow the same here.

**Comment by prakhargupta1** [comment]
Date picker, text field still break:

https://github.com/mui/mui-toolpad/assets/92228082/dd39a412-799d-4900-8a2b-e26bed883787

File: docs/pages/toolpad/components/date-picker.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+import MarkdownDocs from 'docs/src/modules/components/MarkdownDocs';
+import * as pageProps from '../../../data/toolpad/components/datepicker/date-picker.md?@muiMarkdown';
```
Comment (Janpot) [inline docs/pages/toolpad/components/date-picker.js:3]:
```suggestion
import * as pageProps from '../../../data/toolpad/components/datepicker/date-picker.md?muiMarkdown';
```

File: docs/data/toolpad/components/datepicker/date-picker.md
Comment (Janpot) [inline docs/data/toolpad/components/datepicker/date-picker.md:1]: `docs/data/toolpad/components/datepicker/date-picker.md`
=>
`docs/data/toolpad/components/date-picker/date-picker.md`

File: docs/data/toolpad/components/datepicker/DatePickerDisabled.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import { Stack } from '@mui/material';
+
+const TOOLPAD_PROPS1 = {
+  variant: 'outlined',
+  size: 'small',
+  disabled: true,
+  label: 'Disabled',
+};
+
+const TOOLPAD_PROPS2 = {
```
Comment (Janpot) [inline docs/data/toolpad/components/datepicker/DatePickerDisabled.js:12]: Why these property objects? Can't it be done like in https://github.com/mui/mui-toolpad/pull/2827/files#diff-a961ef6fd4d3a6c4228a46a2b7eb76832b7d49135bc1cc8eef36466f5d1d47bdR8

**Comment by Janpot** [comment]
> Date picker, text field still break:

fix added

**Comment by Janpot** [comment]
> About the missing demo in the list component, I guess we can't add it. So I should just remove it?

I don't know what it would look like. @bharatkashyap if you have an idea, otherwise we can omit.

* Can you add links to the PR preview
* Can you address the [vale comments](https://github.com/mui/mui-toolpad/pull/2827/files#diff-f1c5e28976222900939dd882dbaada3581e351b8365f27496a301908d5c66a43R38)

**Comment by apedroferreira** [comment]
Should we remove `isInvalid` for now as discussed in https://github.com/mui/mui-toolpad/issues/2826? I can remove it if we agree.

**Review by Janpot — APPROVED** [review]
Let's merge, I'll add this in the release

**Pull request merged**